isisd/isisd.c

```diff
@@ -545,6 +546,15 @@ void isis_area_destroy(struct isis_area *area)
 	if (!CHECK_FLAG(im->options, F_ISIS_UNIT_TEST))
 		isis_redist_area_finish(area);
 
+	if (listcount(area->area_addrs) > 0) {
+		addr = listgetdata(listhead(area->area_addrs));
```

Comment:
Shouldn't we check for all area_addrs?

Reply:
All the system IDs of the area addresses within the same area must be the same. I don't think it's necessary to check all the area addresses.

Reply:
Makes sense.